### ISSUE TITLE
help(DEVELOPMENT): can't build project or run dev-app on MacBooks with m1 CPUs

### ISSUE THREAD

**Issue by Splaktar** [angular/components#23983]
### What are you trying to do?

On a device running Apple's new m1 CPU

![Screen Shot 2021-11-18 at 13 57 43](https://user-images.githubusercontent.com/3506071/142479285-bfda7541-602b-41d8-94f5-02ce392bb6b2.png)

1. Build the project
2. Run the dev-app

### What troubleshooting steps have you tried?

Setting my `BAZEL` environment variable as discussed in https://github.com/angular/angular/issues/40498#issuecomment-973159023. This fixed Bazel builds for Framework, but not Components.

### Reproduction

Steps to reproduce:
1. yarn
2. yarn build
3. yarn dev-app

### Environment

- Angular: 13.0.1
- CDK/Material: master
- Browser(s): Chrome 95.0.4638.69 (Official Build) (arm64)
- Operating System (e.g. Windows, macOS, Ubuntu): macOS Monterey 12.0.1 (21A559)

**Comment by devversion** [comment]
As stated offline:

* rules_webtesting has been made compatible with M1
* The shared dev-infra package is also fully compatible 

Though:

* ibazel does not yet support Arm64. I got it working by locally building ibazel though. So it would be trivial landing this upstream.
*  The ConcatJS devserver (used in FW only), is not compatible as it is a native binary. rules_nodejs needs some help here.
* Clang-format (used in FW only) seems to need an update. Looks like a new version is finally available (it was not before; and brew was needed)
* Buildifier (for formatting) I think also needs an update to work with Arm64. Last time I checked it did not have a build available for arm64.
* Webdriver Manager, used in FW and CLI (IIRC) does not work with arm64 either. Time to get off protractor?
* Puppeteer does not pull native Arm64 m1 builds last time I checked. It used x86 and required Rosetta (please do not install! it's hard to get rid of it again)

just a summary of what I found already when I worked with M1

**Comment by pkirchniawy** [comment]
Any news on this issue?

**Comment by Splaktar** [comment]
@pkirchniawy nope, I still get the same errors as above when using the latest `master`. Since this is my primary machine, I consider this to be a P1 rather than a P3.

**Comment by Splaktar** [comment]
I guess that this might be the specific steps that need to be done to use your own locally built `ibazel` on an m1 mac:
https://github.com/bazelbuild/bazel-watcher/blob/master/README.md#compiling-yourself

**Comment by Splaktar** [comment]
Unfortunately, that didn't work for me as the local build failed since the `rules_proto` 404'd and it doesn't seem to like my bazel 4.2.1 installation:
```
Starting local Bazel server and connecting to it...
WARNING: Download from https://mirror.bazel.build/github.com/bazelbuild/rules_proto/archive/9e4c622ba8c2178b71420ed3d14fb8874beee3a5.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
INFO: Analyzed target //ibazel:ibazel (65 packages loaded, 7364 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_splaktar/8fee38e85ee44a13c2ac34d7c1f24c9d/external/com_github_fsnotify_fsevents/BUILD.bazel:3:11: GoCompilePkg external/com_github_fsnotify_fsevents/darwin_amd64_stripped/fsevents%/github.com/fsnotify/fsevents.a failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix darwin_amd64 -src external/com_github_fsnotify_fsevents/fsevents.go -src ... (remaining 31 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
xcrun: error: can't exec '/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_devel_bazel/bazel/work/bazelwrap/cc' (errno=No such file or directory)
Error in child process '/usr/bin/xcrun'. 71
compilepkg: error running subcommand: exit status 2
Target //ibazel:ibazel failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 62.041s, Critical Path: 3.16s
INFO: 50 processes: 24 internal, 26 darwin-sandbox.
FAILED: Build did NOT complete successfully
$ bazel --version
bazel 4.2.1- (@non-git)
```
What version of bazel does this build expect?

It looks like this `ibazel` piece is blocked by
- https://github.com/bazelbuild/bazel-watcher/issues/482
- https://github.com/bazelbuild/bazel-watcher/issues/493

These issues (and an associated issue in a dependency) have been open since November 2021 and there doesn't appear to be anyone working on them.

**Comment by Splaktar** [comment]
There is a PR open against bazel-watcher to add m1 mac support: https://github.com/bazelbuild/bazel-watcher/pull/496. It could use some reviews.

**Comment by devversion** [comment]
Also wanted to mention that I will be using M1 only for the next couple of weeks/and potentially months (as I had to move without my Windows machine). So I'm sure we'll be able to get to this very soon. Still want to mention though that I think the majority of workflows should work. The key really is currently using Bazel globally, and avoiding ibazel (or using a custom build)

**Comment by pkirchniawy** [comment]
I was able to build ibazel locally based on the pr branch, but when running ´yarn dev-app´ it is still using ibazel from the npm package installed. Even after setting ibazel into the PATH and running the ibazel  it shows me `zsh: command not found: ibazel`. 

echo $PATH gives me this paths: 
/Users/user/Code/m1-bazel-watcher/bazel-bin/ibazel/
/Users/user/Code/m1-bazel-watcher/bazel-bin/ibazel/darwin_arm64_pure_stripped

Any suggestions?

**Comment by devversion** [comment]
@pkirchniawy you could run ibazel directly. e.g. `ibazel run //src/dev-app:devserver`

**Comment by pkirchniawy** [comment]
> @pkirchniawy you could run ibazel directly. e.g. `ibazel run //src/dev-app:devserver`

Thanks! I've already tried that. But ibazel is not a valid command. After compiling it locally, I struggle with setting the PATH right. (https://github.com/bazelbuild/bazel-watcher#compiling-yourself). So when I type ibazel run, it tells me command not found: ibazel.

**Comment by devversion** [comment]
interesting, that indicates that `ibazel` is not part of the `PATH` properly. I don't know to what you have set the `PATH`, but usually it should include a path pointing to the directory containing the `ibazel` executable binary.

**Comment by pkirchniawy** [comment]
> echo $PATH gives me this paths:
> /Users/user/Code/m1-bazel-watcher/bazel-bin/ibazel/
> /Users/user/Code/m1-bazel-watcher/bazel-bin/ibazel/darwin_arm64_pure_stripped

**Comment by Splaktar** [comment]
`bazel-watcher` has been updated in [0.16.2](https://github.com/bazelbuild/bazel-watcher/releases/tag/v0.16.2) to support m1 macs! 🚀

**Comment by Splaktar** [comment]
If I bump

https://github.com/angular/components/blob/3deca241f994cab5837909a4410256a9e0541cda/package.json#L89

to `0.16.2`, I get farther but now I see this:
```
$ yarn dev-app
yarn run v1.22.17
$ ibazel run //src/dev-app:devserver
iBazel [1:57PM]: You are using an experimental filesystem watcher. If you would like to disable that, please set the environment variable
        IBAZEL_USE_LEGACY_WATCHER=1
iBazel [1:57PM]: Error getting Bazel info exit status 2
iBazel [1:57PM]: Querying for files to watch...
iBazel [1:57PM]: Error getting Bazel info exit status 2
Error finding remote repositories directory: exit status 2
iBazel [1:57PM]: Error getting Bazel info exit status 2
Error finding remote repositories directory: exit status 2
iBazel [1:57PM]: Running //src/dev-app:devserver
Loading: 
Loading: 0 packages loaded
ERROR: Failed to load Starlark extension '@npm//@bazel/protractor:package.bzl'.
Cycle in the workspace file detected. This indicates that a repository is used prior to being defined.
The following chain of repository dependencies lead to the missing definition.
 - @npm
 - @nodejs_darwin_arm64
This could either mean you have to add the '@nodejs_darwin_arm64' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing definition earlier in your WORKSPACE file.
ERROR: cycles detected during target parsing
INFO: Elapsed time: 0.120s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
iBazel [1:57PM]: Error running Bazel exit status 1
error Command failed with exit code 4.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by kevinpauer** [comment]
Hey @Splaktar try changing "@bazel/ibazel": "0.15.10" to "@bazel/ibazel": "0.16.2" in your package.json. This worked for me. :)

**Comment by kevinpauer** [comment]
Had the same issue @Splaktar. Made it work for me following these steps.
1. After forking the most recent angular/components release, I first had to downgrade me node version to 16.x.
2. Followed [this guide](https://github.com/bazelbuild/bazel-watcher#compiling-yourself) to compile ibazel on my device (honestly not sure if it did anything)
3. Bumped up the ibazel version in my package.json from "@bazel/ibazel": "0.15.10" to "@bazel/ibazel": "0.16.2", this version seems to support the M1
4. `yarn` then `yarn dev-app` first time did somehow fail, started again and it ran through

Hope this helps.

**Comment by devversion** [comment]
@RobertAKARobin thx for the details. I think your branch is outdated and would need a rebase. We updated both the bazel-watcher and Bazel already to an M1 compatible version.

**Comment by RobertAKARobin** [comment]
That worked, thanks!
